src/__swaps__/screens/Swap/components/ExchangeRateBubble.tsx

```diff
@@ -1,34 +1,135 @@
-    return `1,624.04 ${SwapInputController.assetToBuy.value?.symbol}`;
-  });
+        // try to set price immediately
+        const price = priceForAsset({
+          asset: current.assetToSell,
+          assetType: 'assetToSell',
+          assetToSellPrice: SwapInputController.assetToSellPrice,
+          assetToBuyPrice: SwapInputController.assetToBuyPrice,
+        });
+
+        if (price) {
+          assetToSellPrice.value = price;
+        }
+      }
+
+      if (current.assetToBuy && (!previous?.assetToBuy || current.assetToBuy !== previous.assetToBuy)) {
+        assetToBuySymbol.value = current.assetToBuy.symbol;
+
+        // try to set price immediately
+        const price = priceForAsset({
+          asset: current.assetToBuy,
+          assetType: 'assetToBuy',
+          assetToSellPrice: SwapInputController.assetToSellPrice,
+          assetToBuyPrice: SwapInputController.assetToBuyPrice,
+        });
 
-  // TODO: Do proper exchange rate calculation once we receive the quote
+        if (price) {
+          assetToBuyPrice.value = price;
+        }
+      }
 
-  // TODO: This doesn't work when assets change, figure out why...
-  if (!assetToSellLabel.value || !assetToBuyLabel.value) return null;
+      if (current.assetToSell && current.assetToBuy) {
+        runOnJS(SwapInputController.fetchAssetPrices)({
+          assetToSell: current.assetToSell,
+          assetToBuy: current.assetToBuy,
+        });
+      }
+
+      if (current.assetToSellPrice && (!previous?.assetToSellPrice || current.assetToSellPrice !== previous.assetToSellPrice)) {
+        assetToSellPrice.value = current.assetToSellPrice;
+      }
+
+      if (current.assetToBuyPrice && (!previous?.assetToBuyPrice || current.assetToBuyPrice !== previous.assetToBuyPrice)) {
+        assetToBuyPrice.value = current.assetToBuyPrice;
+      }
+
+      if (assetToSellPrice.value && assetToBuyPrice.value) {
+        switch (exchangeRateIndex) {
+          // 1 assetToSell => x assetToBuy
+          case 0: {
+            // TODO: Make the decimal points pretty on the rate conversion
+            const rate = assetToSellPrice.value / assetToBuyPrice.value;
+            fromAssetText.value = `1 ${assetToSellSymbol.value}`;
+            toAssetText.value = `${rate} ${assetToBuySymbol.value}`;
+            break;
+          }
+          // 1 assetToBuy => x assetToSell
+          case 1: {
+            // TODO: Make the decimal points pretty on the rate conversion
+            const rate = assetToBuyPrice.value / assetToSellPrice.value;
+            fromAssetText.value = `1 ${assetToBuySymbol.value}`;
+            toAssetText.value = `${rate} ${assetToSellSymbol.value}`;
+            break;
+          }
+          // assetToSell => native currency
+          case 2: {
+            fromAssetText.value = `1 ${assetToSellSymbol.value}`;
+            toAssetText.value = `$${assetToSellPrice.value.toLocaleString('en-US', {
```

Comment:
we shouldnt be using toLocaleString since it doenst play with with the rest of the supported currencies, we have a JS util might just need to write it as a worklet

Reply:
Added a new ticket: https://linear.app/rainbow/issue/APP-1346/switch-tolocalestring-to-support-other-nativecurrencies